### PR TITLE
Update action.mdx

### PR DESCRIPTION
**Pull request by madaxen86**
Added documentation for revalidation of cached functions with actions and revalidate helper.

### CODE REVIEW

File: src/routes/solid-router/reference/data-apis/action.mdx [hunk 1]
```diff
@@ -100,3 +101,69 @@ type Submission<T, U> = {
 const submissions = useSubmissions(action, (input) => filter(input));
 const submission = useSubmission(action, (input) => filter(input));
 ```
+##  Revalidate cached functions
+
+###  Revalidate all (default)
+By default all cached function will be revalidated wether the action does not return or return a "normal" response.
```
Comment (atilafassina) [inline src/routes/solid-router/reference/data-apis/action.mdx]:
```suggestion
By default all cached functions will be revalidated wether the action does not return or return a "normal" response.
```

File: src/routes/solid-router/reference/data-apis/action.mdx [hunk 2]
```diff
@@ -100,3 +101,69 @@ type Submission<T, U> = {
...
+```jsx
+
+const deleteTodo = action(async (formData: FormData) => {
+  const id = Number(formData.get("id"))
+  await api.deleteTodo(id)
+  // return new  Response("success",  {  status:  200  });
+  // ...
```
Comment (atilafassina) [inline src/routes/solid-router/reference/data-apis/action.mdx]: It's preffered that snippets are copy-pasteable. And in this case I think the return would be ideal to have. What do you think?

```suggestion
  // ...
  return new  Response("success",  {  status:  200  });
```

File: src/routes/solid-router/reference/data-apis/action.mdx [hunk 3]
```diff
@@ -100,3 +101,69 @@ type Submission<T, U> = {
...
+})
+```
+
+### Revalidate specific cached keys
+
+By returning a response with solid router's `json` or `reload` helpers you can pass a key / keys to the revalidate prop as the second argument of solid's router json response helper.  You can either pass as string directly or use the cached functions `key` or `keyFor` props
```
Comment (atilafassina) [inline src/routes/solid-router/reference/data-apis/action.mdx]:
```suggestion
By returning a response with solid-router's `json` or `reload` helpers you can pass a key / keys to the revalidate prop as the second argument of the json response helper.
You can either pass as `string` directly or use the cached functions `key` or `keyFor` props.

File: src/routes/solid-router/reference/data-apis/action.mdx [hunk 4]
```diff
@@ -100,3 +101,69 @@ type Submission<T, U> = {
...
+  return json(
+  `deleted ${id}`,
+  { revalidate: ["getAllTodos",getTodos.key, getTodoByID.keyFor(id)]}
+  )
+  //or
+  return reload({ revalidate: ["getAllTodos",getTodos.key, getTodoByID.keyFor(id)]})
+})
+```
+
+### No revalidation
```
Comment (atilafassina) [inline src/routes/solid-router/reference/data-apis/action.mdx]: Imho, "prevent" seems to convey better the message than simply "no" in this case.

```suggestion
Prevent revalidation
```

File: src/routes/solid-router/reference/data-apis/action.mdx [hunk 5]
```diff
@@ -100,3 +101,69 @@ type Submission<T, U> = {
...
+To opt out of any revalidation you can pass any string to revalidate which is not a key of any cached function.
```
Comment (atilafassina) [inline src/routes/solid-router/reference/data-apis/action.mdx]:
```suggestion
To opt out of any revalidation you can pass any `string` to revalidate which is not a key of any cached function.
```

File: src/routes/solid-router/reference/data-apis/action.mdx [hunk 6]
```diff
@@ -100,3 +101,69 @@ type Submission<T, U> = {
...
+```jsx
+const deleteTodo = action(async (formData: FormData) => {
+  const id = Number(formData.get("id"))
+  await api.deleteTodo(id)
+  return json(`deleted ${id}`,{ revalidate: "nothing" })
+  //or
+  return reload({ revalidate: "nothing" })
```
Comment (atilafassina) [inline src/routes/solid-router/reference/data-apis/action.mdx]:
```suggestion
  // returns a `json` without revalidating the action.
  return json(`deleted ${id}`,{ revalidate: "nothing" })

  // or reload the route without revalidating the request.
  // return reload({ revalidate: "nothing" })
```

If showing alternatives, I'd propose we point out the differences between them explicitly.
(the wording is just a suggestion, I'm happy if you go with something else)

File: src/routes/solid-router/reference/data-apis/action.mdx [hunk 7]
```diff
@@ -100,3 +101,69 @@ type Submission<T, U> = {
...
+})
+```
+
+###  Revalidate without action
+Cached functions can also be revalidated by the  `revalidate` helper.
+```jsx
+revalidate([getTodos.key, getTodoByID.keyFor(id)])
+```
+This is also great if you want to set your own "refresh" interval e.g. poll data every 30 seconds .
```
Comment (atilafassina) [inline src/routes/solid-router/reference/data-apis/action.mdx]:
```suggestion
This is also great if you want to set your own "refresh" interval e.g. poll data every 30 seconds.
```

File: src/routes/solid-router/reference/data-apis/action.mdx [hunk 8]
```diff
@@ -100,3 +101,69 @@ type Submission<T, U> = {
...
+	
+	const todos = createAsync(() => getTodos())
+	
+	onMount(() => {
+		//30 second polling
+		const interval = setInterval(() => revalidate(getTodos.key),1000 * 30)
+		onCleanup(() => clearInterval(interval))
+	})
+	
+	...
```
Comment (atilafassina) [inline src/routes/solid-router/reference/data-apis/action.mdx]: Use the same formatting as the previous snippets and comment out what isn't valid code.

```suggestion
	// ...
```

**Review by atilafassina — CHANGES_REQUESTED** [review]
Thanks for this!! Great work!! 🏆

I made a few minor suggestions, feel free to take them entirely, partially, or reply :)

Looking forward to merge this!

**Comment by madaxen86** [comment]
Good suggestions.
I've also added the `redirect` helper as an option.

**Review by atilafassina — APPROVED** [review]

**Pull request merged**